### ISSUE TITLE
Download of Windows installer (MSI) fails 3 times before downloading

### ISSUE THREAD

**Issue by JeeWeetje** [Azure/azure-docs-cli-python#607]
I noticed strange behavior using Firefox on two different machines while downloading the .msi file from https://aka.ms/InstallAzureCliWindows. It is a redirect to an Azure blob storage which will fail the first 3 times serving the file as download. The fourth time it will download after all.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8b7bd326-07b2-caaa-7155-5852fc2dfa3f
* Version Independent ID: c44d0547-d5cd-8934-00f8-79e2f4df53ac
* [Content](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)
* [Content Source](https://github.com/Azure/azure-docs-cli-python/blob/live/docs-ref-conceptual/install-azure-cli.md)
* Service: multiple

**Comment by micahmckittrick** [comment]
Thanks for reaching out. I have attempted to repro this issue however when following that link I see no issues downloading it on Firefox. I also had a few coworkers attempt the link and they were also able to download first try. 

Some suggestions that you could try: 

1) Make sure FireFox is fully up to date
2) Clear Cache
3) Try downloading the link from a private window
4) Turn off pop up blocker
5) Uncheck "Block Dangerous Downloads". Note, you should turn this on again but turn it off for troubleshooting purposes.

**Comment by chriswhellams** [comment]
Hi, I have the same problem using Google Chrome. Download starts then I get "Network Error" after ~30 secs

**Comment by wolffhaven** [comment]
I too have same problem.

**Comment by gaetanop84** [comment]
I have the same problem with Chrome.
If I try to download this one, it works:
https://azurecliprod.blob.core.windows.net/msi/azure-cli-2.0.25.msi

**Comment by AmirSasson** [comment]
i have the same problem, when clicking on *Download the MSI installer* the download failed after couple of seconds..
_Failed - Network error_

**Comment by sptramer** [comment]
I've alerted the CLI team so that they can investigate this problem.

**Comment by micahmckittrick** [comment]
@sptramer @derekbekoe Can either of you repro this issue? The file downloads just fine for me but obviously a few people are hitting this wall.

**Comment by derekbekoe** [comment]
@sptramer and I have discussed offline.
@sptramer can you comment on what we discussed?

**Comment by derekbekoe** [comment]
The most recent MSI will always be available at so no need for the aka.ms link which seemed to cause issues:
https://azurecliprod.blob.core.windows.net/msi/azure-cli-latest.msi

**Comment by sptramer** [comment]
The links on the Windows installer page have been updated to ensure that they are pointed directly at the production latest MSI, not through a redirect.

https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-windows?view=azure-cli-latest

**Comment by savasayik** [comment]
Cant download too. I have the same problem with Chrome, Firefox and Edge.

**Comment by micahmckittrick** [comment]
Still downloads perfectly the first time in every browser I try. What error do you get?

**Comment by lisandro101** [comment]
Same issue here, with Chrome impossible but with Edge after several attempts it worked

**Comment by sptramer** [comment]
@MarkDerman @lisandro101 @savasayik - Are any of you blocked by a firewall, proxy, or region setting form being able to reach https://azurecliprod.blob.core.windows.net/msi/azure-cli-latest.msi ? This is no longer a redirect so there is not a possibility of 404 or URL mangling.

**Comment by derekbekoe** [comment]
In addition to the question from @sptramer , what locations/countries are you in?

**Comment by beppler** [comment]
I can't download using chrome too, but it worked when I used:

```powershell
Invoke-WebRequest -OutFile azure-cli-latest.msi https://azurecliprod.blob.core.windows.net/msi/azure-cli-latest.msi
```

PS: I'm from Curitiba, Brazil.

**Comment by n-alex-white** [comment]
The problem still exists - using chrome I get Failed - network error after about 6 seconds of downloading, every time. Very frustrating. No problems with anything on any other site. Just this.

**Comment by VishDev12** [comment]
I'm facing the same issue too. The problem is definitely not with our networks. I'm able to download other large files easily. Does anyone have a mirror to the file?

**Comment by derekbekoe** [comment]
https://azurecliprod.blob.core.windows.net/msi/azure-cli-latest.msi

If anyone can use the download link above and while Chrome dev tools is open on the Network tab and send a screenshot or details of the server response, that'd be useful.

**Comment by sptramer** [comment]
Reopening this issue until we can confirm what the problem is and whether or not the fix is on engineering/distribution, or docs.

**Comment by n-alex-white** [comment]
I was able to install it by using chocolatey

**Comment by jayeshdshah** [comment]
Thanks @lisandro101.. worked like a charm with Edge..

I also tried 10+ times with chrome but didn't worked but with Edge, in second try download was successful...

**Comment by jkewley** [comment]
It looks like the signing cert has been revoked  by Microsoft.

**Comment by derekbekoe** [comment]
@jkewley It works here. What's your issue?

**Comment by jkewley** [comment]
I just downloaded again and see this:
![image](https://user-images.githubusercontent.com/1148930/36806654-ba533fa0-1c8e-11e8-8274-6331332b3aca.png)

**Comment by jkewley** [comment]
yes, sorry for side-jacking - I thought the downloads could be impacted by the invalid signing cert

**Comment by derekbekoe** [comment]
@jkewley Please open a separate issue at https://github.com/Azure/azure-cli/.
I see this:
<img width="792" alt="screen shot 2018-02-28 at 11 01 13 am" src="https://user-images.githubusercontent.com/16448634/36807057-c6a8ab04-1c76-11e8-8590-b54178fe8984.png">

**Comment by derekbekoe** [comment]
As a workaround, for the 2.0.28 release, I've added the MSI to our GitHub release page.
https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.0.28
Download link: https://github.com/Azure/azure-cli/releases/download/azure-cli-2.0.28/azure-cli-2.0.28.msi

Let me know if you still have issues with that download. I would be interested to know either way.
note: right now, there's no guarantee that we'll publish future releases to GitHub also.

**Comment by jpdoiron** [comment]
Also would failed after 3 sec with chrome, but worked with internet explorer.

**Comment by marckassay** [comment]
I had this issue last night when I used Chrome. So created a [PowerShell script](https://github.com/marckassay/XAz/blob/0.0.1/docs/Invoke-AzureCLIDownload.md) to download it which seems to work.
